Sway Verifier generator for bb 0.87.0

ewynx opened the pull request:
This PR add the command `write_sway_verifier` to `bb`. It is targeting a branch made off the code at tag `v0.87.0`, since the Sway UltraHonk verifier aligns with the Solidity verifier of that version (nargo 1.0.0-beta.11, bb 0.87.0). 

Implements the UltraHonk verifier in Sway, following the example of UltraHonk verifier in Solidity. One difference is that Sway can't work with dynamic array length (defined by constants), whereas Solidity can, so all those values are hardcoded in Sway when they are not in Solidity. This is also why the generation of the Sway code takes an additional step; replacing all occurrences of `${PUB_INPUT_SIZE}` with the actual value. 

Steps to try this out locally: https://gist.github.com/ewynx/54f73ecc05a130084dae620138f85b45

ewynx committed:
feat: add Sway verifier generator aligned with bb v0.87.0

ewynx committed:
Add contract to Sway verifier + cleanup of comments

ewynx committed:
Info msg fix

man2706kum commented on barretenberg/cpp/src/barretenberg/dsl/acir_proofs/honk_sway_contract.hpp:
```diff
@@ -0,0 +1,1697 @@
+}
+
+impl MyContract for Contract {
+    fn verify_proof(proof: Proof, public_inputs: [u256; ${PUB_INPUT_SIZE}]) -> bool {
```
This will error when PUB_INPUT_SIZE is 0. The contract should also support for 0 public inputs

man2706kum replied:
```
fn main(x: Field) {
    assert(x != 0);
}
```
this noir program has 0 public inputs

ewynx replied:
great catch, thank you!

man2706kum reviewed (commented):
Comment: https://github.com/hashcloak/sway-zkp-verifiers/pull/4#pullrequestreview-3228675952

ewynx committed:
PR fixes: Support public input size 0 and remove pub before fns.

ewynx commented:
@man2706kum I fixed the 2 comments. 

To support public input size 0 I decided to add more magic to the templating. Another option might have been to use a `Vec`, but I think this approach is preferred since an array will do for the actual functionality. A `Vec` would mainly be to make the template function cleaner, but I don't think that is the most important. 

Furthermore, I removed all `pub`s. Seems like it's not necessary at all, not even in the contract itself!

man2706kum reviewed (approved):
LGTM!
